src/sesclient.native/README.md

```diff
@@ -5,14 +5,17 @@ This command line executable demonstrates use of the [native-code software entit
 
 ## Working with sesclient.native
+
 This executable is intended to be used in conjunction with the [sestest command line utility](../sestest) to generate tokens.
 
 The executable expects the following parameters (in any order):
 
-| Parameter            | Required  | Definition                                                                                                                         |
-| -------------------- | --------- | ---------------------------------------------------------------------------------------------------------------------------------- |
-| --url                | Mandatory | The URL at which the server should process requests. <br/> **Note**: must start with `https:`.                                      |
-| --thumbprint         | Mandatory | Thumbprint of a certificatein the server's SSL certificate chain. <br/> **Note**: cannot be the thumbprint of the root certificate. |
-| --common-name        | Mandatory | The common name of the certificate indicated by `--thumbprint`.                                                                    |
-| --token              | Mandatory | The token as returned by `sestest generate`.                                                                                       |
-| --application        | Mandatory | Unique identifier for the application being requested.                                                                             |
+|   Parameter   | Required  |                                                                   Definition                                                                    |
+| ------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------- |
+| --url         | Mandatory | The URL of the server that will process our request. <br/> **Note**: must start with `https:`.                                                  |
+| --token       | Mandatory | The token as returned by `sestest generate`.                                                                                                    |
+| --application | Mandatory | Unique identifier for the application being requested.                                                                                          |
+| --thumbprint  | Optional  | Thumbprint of an additional certificate to accept in the TLS certificate chain. <br/> **Note**: cannot be the thumbprint of a root certificate. |
+| --common-name | Optional  | The common name of the certificate indicated by `--thumbprint`.                                                                                 |
+
+The library includes a set of well known Microsoft intermediate certificates that are always accepted. The `--thumbprint` and `--common-name` options allow another certificate to be added into the set of acceptable certificates, allowing for local testing in environments without access to those Microsoft certificates.
```

Comment:
The docs imply this is specifically for testing tokens created by `sestest generate` -- would these ever be signed by a Microsoft intermediate certificate?  Are we saying that `sesclient` will hunt back along the signing chain until it meets a Microsoft intermediate cert (e.g. for a token exfiltrated from a Batch compute node) or the custom cert (for a token created using `sestest generate`)?  Because this is done by the client rather than the service, correct?  The certs we're talking about in this paragraph are the signing certs, not HTTPS auth certs, correct?  (Forgive me if this is all covered in other parts of the docs not included in this PR... I can be a Beard of Very Little Brain at times...)

Reply:
One supported scenario for `sesclient` is in a production environment, as a diagnostic tool - this allows us to cut out a third party application and verify whether the end-to-end handshake for software entitlements is working properly.

But, the context here is about securing the HTTPS connection to the server. We require the connection to be secured by a certificate tracing through one of the known intermediate certificates so that a MITM attack cannot happen.